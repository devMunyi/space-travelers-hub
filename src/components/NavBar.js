import { NavLink } from 'react-router-dom';

const NavBar = () => {
  const links = [
    { id: 1, path: '/', text: 'Rockets' },
    { id: 2, path: '/missions', text: 'Missions' },
    {
      id: 3,
      path: '/profile',
      text: 'Profile',
    },
  ];

  return (
    <nav
      style={{ backgroundColor: '#fff' }}
      className="mb-2 d-flex justify-content-between align-items-center container-fluid border-bottom"
    >
      <div style={{ width: '50%' }} className="nav-left-wrapper">
        <div className="fs-3">
          <NavLink className="link-item brand mx-auto" to="/">
            <img height="40" width="50" src="planet.png" alt="LOGO" /> Space
            Traveler's Hub
          </NavLink>
        </div>
      </div>
      <ul
        style={{ width: '50%' }}
        className="nav d-flex justify-content-around fs-6"
      >
        {links.map((link) => (
          <li key={link.id}>
            <NavLink className={`link-item`} to={link.path}>
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default NavBar;
