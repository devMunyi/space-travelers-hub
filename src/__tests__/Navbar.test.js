import { screen, render } from '@testing-library/react';

import NavBar from '../components/NavBar';
import { BrowserRouter as Router } from 'react-router-dom';

const MockedNavBar = () => (
  <Router>
    <NavBar />
  </Router>
);

describe('Navbar', () => {
  it('should render navbar correctly', () => {
    render(<MockedNavBar />);
    const rocketsText = screen.getByText(/rockets/i);

    expect(rocketsText).toMatchSnapshot();
  });
});
