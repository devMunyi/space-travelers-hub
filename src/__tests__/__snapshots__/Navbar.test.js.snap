// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render navbar correctly 1`] = `
<a
  aria-current="page"
  class="link-item active"
  href="/"
>
  Rockets
</a>
`;

exports[`Navbar should render navbar correctly 2`] = `
<a
  class="link-item"
  href="/missions"
>
  Missions
</a>
`;
